### ISSUE TITLE
Batch size < 5 does not reduce training loss

### ISSUE THREAD

**Issue by gaetanoantonicchio** [AlexPasqua/NNs-from-scratch#36]
Bug loss function when batch size is less than 5

**Comment by AlexPasqua** [comment]
@gaetanoantonicchio
Intanto: si nota che con un batch piccolo, il gradiente medio di quel batch sia 0 da un certo punto in poi

**Comment by AlexPasqua** [comment]
@gaetanoantonicchio 
Il problema è la ReLU, che azzera il gradiente. Sostituita con la sigmoid dà i seguenti grafici (ovviamente manca lo shuffling del dataset)

**Comment by AlexPasqua** [comment]
@gaetanoantonicchio 
Il problema è la ReLU, che azzera il gradiente. Sostituita con la sigmoid dà i seguenti grafici (ovviamente manca lo shuffling del dataset)

10 epochs:
![image](https://user-images.githubusercontent.com/33461106/102631535-24778800-414e-11eb-8a97-89da8a98ab40.png)

50 epochs:
![image](https://user-images.githubusercontent.com/33461106/102631500-175a9900-414e-11eb-87a0-309683b97262.png)

150 epochs:
![image](https://user-images.githubusercontent.com/33461106/102631612-3bb67580-414e-11eb-8616-77429c2e8bd0.png)

300 epochs:
![image](https://user-images.githubusercontent.com/33461106/102631658-4a9d2800-414e-11eb-89eb-918cc20e20de.png)

500 epochs:
![image](https://user-images.githubusercontent.com/33461106/102631803-7b7d5d00-414e-11eb-89f3-315b93a3bd82.png)
